tests/helm/README.md

```diff
@@ -1,68 +1,80 @@
+```bash
+make validate-helm
+```
 
-- **S3 Endpoint URL Requirement**: Ensures that the S3 endpoint URL is properly marked as required in the Helm chart
-- _(More validations can be added over time)_
+This runs the validation script which performs:
 
-### Usage
+- Helm template rendering validation
+- YAML syntax checking
+- Required field validation
+- Security policy compliance
 
-```bash
-# Run all validations
-./validate_charts.sh
+## Test Structure
 
-# Run a specific validation
-./validate_charts.sh validate_s3_endpoint_required
+### Chart Validation (`validate_charts.sh`)
 
-# Show help
-./validate_charts.sh --help
-```
+Core validation script that:
 
-## Adding New Validations
+- Validates Helm chart syntax and structure
+- Checks for required values and configurations
+- Verifies generated manifest correctness
+- Tests default and custom value scenarios
 
-To add a new validation test:
+### Template Tests
 
-1. Add a new function to `validate_charts.sh` following the pattern of existing validations
-2. Add the new function to the list of validations in the `main()` function
-3. Update the usage information in the `usage()` function
+Tests that validate generated Kubernetes manifests:
 
-Example:
+- Service account and RBAC configurations
+- Pod security contexts and resource limits  
+- Volume and mount configurations
+- Network policies and service definitions
 
-```bash
-# New validation function
-validate_my_new_check() {
-  local chart_dir="$CHARTS_DIR/scality-mountpoint-s3-csi-driver"
-  
-  # Add validation logic here
-  # Return 0 for success, non-zero for failure
-  
-  return 0
-}
-
-# Then in the main() function, add:
-run_validation "My new check description" validate_my_new_check || ((errors++))
-```
+## Test Categories
+
+### Syntax Validation
 
-## Integration with CI
+- Helm chart YAML syntax
+- Template function usage
+- Value reference validation
 
-The validation script can be integrated with CI by adding a Makefile target:
+### Security Validation  
 
-```makefile
-.PHONY: validate-helm
-validate-helm:
-	@echo "Validating Helm charts..."
-	@tests/helm/validate_charts.sh
+- Pod security contexts
+- Service account permissions
+- Network policy configurations
+- Resource limit enforcement
+
+### Functional Validation
+
+- Required field presence
+- Default value application
+- Custom value override behavior
+- Conditional template rendering
+
+## Configuration
+
+Tests use the chart's default values but can be customized:
+
+```bash
+# Test with custom values
+helm template test-release ./charts/scality-mountpoint-s3-csi-driver \
+  --values custom-values.yaml \
+  --debug
 ```
 
-Then add `validate-helm` to the matrix of tests in `.github/workflows/code-validation.yaml`.
+## Adding New Tests
 
-## Why These Tests Matter
+1. Add validation logic to `validate_charts.sh`
+2. Include test cases for new chart features
+3. Update this documentation with new test descriptions
+4. Ensure tests run in CI environment
```

Comment:
Your linter should complain when you write a numered list like this

```suggestion
1. Add validation logic to `validate_charts.sh`
1. Include test cases for new chart features
1. Update this documentation with new test descriptions
1. Ensure tests run in CI environment
```

Reply:
Resolved per the comment belowe discussion.